### PR TITLE
[ACM-4260]: Creating ARM NodePools on AWS hosted clusters

### PR DESCRIPTION
**Pull request by lahinson**
Related Jira: https://issues.redhat.com/browse/ACM-4260

This PR adds documentation about creating ARM NodePools in hosted clusters on AWS.

### CODE REVIEW

File: clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc [hunk 1]
```diff
@@ -44,4 +44,95 @@ hcp create nodepool aws \
...
+You can schedule application workloads (`NodePool` objects) on 64-bit ARM and AMD64 from the same hosted control plane. To do so, you define the `arch` field in the `NodePool` specification to set the required processor architecture for the `NodePool` object. The valid values for the `arch` field are as follows:
+
+* `arm64`
+* `amd64`
+
+If you do not specify a value for the `arch` field, the `amd64` value is used by default.
+
+To create an ARM `NodePool` object on a hosted cluster on AWS, complete the following steps:
+
+. Ensure that you have a multi-architecture image for the `HostedCluster` custom resource to use. You can access multi-architecture nightly images at link:https://multi.ocp.releases.ci.openshift.org/[https://multi.ocp.releases.ci.openshift.org/].
```
Comment (bryan-cox) [inline clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc:61]:
```suggestion
Ensure that you have a multi-architecture image for the `HostedCluster` custom resource to use. You can access multi-architecture nightly images at link:https://multi.ocp.releases.ci.openshift.org/[https://multi.ocp.releases.ci.openshift.org/].
```

File: clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc [hunk 2]
```diff
@@ -44,4 +44,95 @@ hcp create nodepool aws \
...
++
+----
+apiVersion: hypershift.openshift.io/v1beta1
+kind: NodePool
+metadata:
+  creationTimestamp: null
+  name: hypershift-arm-us-east-1a
+  namespace: clusters
+spec:
+  arch: arm64
```
Comment (bryan-cox) [inline clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc]: Actually arch will be `amd64` at this point before the sed command.

File: clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc [hunk 3]
```diff
@@ -44,4 +44,95 @@ hcp create nodepool aws \
...
+  clusterName: hypershift-arm
+  management:
+    autoRepair: false
+    upgradeType: Replace
+  nodeDrainTimeout: 0s
+  platform:
+    aws:
+      instanceProfile: hypershift-arm-2m289-worker
+      instanceType: m6g.large
```
Comment (bryan-cox) [inline clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc]: Actually instanceType will be `m5.large` at this point before the sed command.

File: clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc [hunk 4]
```diff
@@ -44,4 +44,95 @@ hcp create nodepool aws \
...
+        id: subnet-02c74cf1cf1e7413f
+    type: AWS
+  release:
+    image: quay.io/openshift-release-dev/ocp-release-nightly@sha256:390a33cebc940912a201a35ca03927ae5b058fbdae9626f7f4679786cab4fb1c
+  replicas: 3
+status:
+  replicas: 0
+----
+
+. Modify the `arch` and `instanceType` values in the YAML file by entering the following command. In the command, the ARM instance type is `m6g.large`, but any ARM instance type works:
```
Comment (bryan-cox) [inline clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc:124]:
```suggestion
Modify the `arch` and `instanceType` values in the YAML file by entering the following command. In the command, the ARM instance type is `m6g.large`, but any ARM instance type works:
```

File: clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc [hunk 5]
```diff
@@ -44,4 +44,95 @@ hcp create nodepool aws \
...
+
++
+----
+sed 's/arch: amd64/arch: arm64/g; s/instanceType: m5.large/instanceType: m6g.large/g' arm_nodepool_spec.yml > temp.yml && mv temp.yml arm_nodepool_spec.yml
+----
+
+. Apply the rendered YAML file to the hosted cluster by entering the following command:
```
Comment (bryan-cox) [inline clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc:131]:
```suggestion
Apply the rendered YAML file to the hosted cluster by entering the following command:
```

**Review by bryan-cox — COMMENTED** [review]
Maybe those . (dots) are supposed to be there, but I added comments for them just in case.

File: clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc (hunk 1 above)
Comment (lahinson) [reply clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc:61]: @bryan-cox The dots are supposed to be there. That's how you indicate a numbered step in asciidoc :)

Comment (cadawson) [reply clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc:61]: Colon after "at"?

File: clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc [hunk 6]
```diff
@@ -44,4 +44,95 @@ hcp create nodepool aws \
...
+
++
+----
+hcp create nodepool aws --cluster-name $CLUSTER_NAME --name $ARM64_NODEPOOL_NAME --node-count=$NODEPOOL_REPLICAS --render > arm_nodepool_spec.yml
+----
+
++
+The command creates a YAML file that specifies the CPU architecture for the `NodePool` object, as shown in the following example:
+
++
```
Comment (cadawson) [inline clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc:89]:
```suggestion
+[source,yaml]
```

**Review by cadawson — APPROVED** [review]
Looks good. A couple of suggestions. Thanks!

File: clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc (hunk 6 above)
Comment (lahinson) [reply clusters/hosted_control_planes/hosted_cluster_arm_aws.adoc:89]: @cadawson Good question about the colon after "at". According to [IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=elements-web-ip-email-addresses#referring-to-web-addresses), we don't use a colon between the verb or preposition and the link text.

**Pull request merged**